### ISSUE TITLE
Enforce await on async utilities 

### ISSUE THREAD

**Issue by cdagli** [testing-library/eslint-plugin-testing-library#49]
Currently https://github.com/Belco90/eslint-plugin-testing-library/blob/master/docs/rules/await-async-query.md rule only supports findBy* and findByAll*. What do you think about enabling it for other async utilities, too? 

https://testing-library.com/docs/dom-testing-library/api-async

```
const ASYNC_QUERIES_REGEXP = /^find(All)?By(LabelText|PlaceholderText|Text|AltText|Title|DisplayValue|Role|TestId)$/;
```

**Comment by thomlom** [comment]
Thanks for suggesting this idea @cdagli! 
This rule currently exists because `findBy` and `findByAll` doesn't (in my opinion) expresses the fact that there are actually asynchronous queries.
However, it's pretty clear that the other async utilities such as `wait` or `waitForElement` are asynchronous. I'm not sure if we would need to enable them 🤔 
Have you encountered a use case where someone forgot to use `await` or `.then` on one of these async utilities? 🙂

**Comment by cdagli** [comment]
Yes, I know couple of people that keeps forgetting to add await these methods 😄 (Myself included) 

Weird stuff starts to happen if you forget to await. And for example if you have a test case that ends up with `waitForElementToBeRemoved` and if you don't await, your test might pass magically.

**Comment by Belco90** [comment]
I just realized eslint is not warning about that (as those async utils are fixed, not as `findBy*` queries which are dynamically generated): I thought it did but it was my IDE! So this is an interesting idea, maybe something like `await-async-utils`?

**Comment by thomlom** [comment]
In the end, forgetting to `await` an async function is a common mistake, not just when using Testing Library's tools. I tried to search if there was a way to prevent this globally but haven't found anything.
It would be cool to have a tool that performs this check when calling async functions.

**Comment by benmonro** [comment]
👍 I would love this rule!

**Comment by Belco90** [comment]
I'm back to handle this issue. Just to clarify, the expected utils checked here would be:

- `wait`
- `waitForElement`
- `waitForDomChange`
- `waitForElementToBeRemoved`

Right?

**Comment by benmonro** [comment]
Might as well include findBy queries as well @Belco90
